src/main/java/eu/darkbot/popcorn/def/AntiPush.java

```diff
@@ -45,15 +65,68 @@ public void setConfig(Config config) {
 
     @Override
     public void tick() {
+        tickDrawFire();
+        tickDeathPause();
+    }
+
+    // updating deathStats on tickStopped() because behaviour not ticked when ship isDead()
+    @Override
+    public void tickStopped() {
+        if (config.DEATH_PAUSE_TIME == 0) return;
+
+        updateDeathStats();
+        if (repairManager.isDead()) hasCountedDeath = false;
+        else {
+            Ship killer = ships.stream()
+                    .filter(s -> s.playerInfo.username.equals(repairManager.getKillerName()))
+                    .findFirst()
+                    .orElse(null);
+            if (!hasCountedDeath && killer != null) {
+                deathStats.computeIfAbsent(killer.id, l -> new ArrayList<>())
+                        .add(Instant.now());
+            }
+            hasCountedDeath = true;
+        }
+    }
+
+    private void tickDrawFire() {
+        if (config.DRAWFIRE_PAUSE_TIME == 0) return;
+
         for (Ship ship : ships) {
             if (!ship.playerInfo.isEnemy() || !ship.hasEffect(EffectManager.Effect.DRAW_FIRE) || !mapManager.isTarget(ship)) continue;
             System.out.println("Pausing bot" +
-                    (config.PAUSE_TIME > 0 ? " for " + config.PAUSE_TIME + " minutes" : "") +
+                    (config.DRAWFIRE_PAUSE_TIME > 0 ? " for " + config.DRAWFIRE_PAUSE_TIME + " minutes" : "") +
                     ", enemy used draw fire");
 
-            Long pauseMillis = config.PAUSE_TIME > 0 ? (long) config.PAUSE_TIME * 60 * 1000 : null;
+            Long pauseMillis = config.DRAWFIRE_PAUSE_TIME > 0 ? (long) config.DRAWFIRE_PAUSE_TIME * 60 * 1000 : null;
             main.setModule(new DisconnectModule(pauseMillis, I18n.get("module.disconnect.reason.draw_fire")));
         }
     }
 
+    private void tickDeathPause() {
+        if (config.DEATH_PAUSE_TIME == 0) return;
+
+        Map.Entry<Integer, List<Instant>> deathEntry = deathStats.entrySet().stream()
+                .filter(e -> e.getValue().size() >= config.MAX_DEATHS)
+                .findFirst()
+                .orElse(null);
+
+        if (deathEntry != null) {
```

Comment:
```suggestion
        if (deathStats.values().stream().anyMatch(v -> v.size() >= config.MAX_DEATHS)) {
```

Reply:
i think I need this, because in the if statement i use deathEntry to print stuff and remove the item from map.

Reply:
nvm I rewrote it using stream().ifPresent() so looks cleaner?